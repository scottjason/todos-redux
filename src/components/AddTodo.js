import React, { useState } from 'react';
import InputField from './shared/InputField';

const AddTodo = props => {
  const [content, setContent] = useState('');
  
  const label = {
    text: 'add to do',
    htmlFor: 'content',
  };

  const onSubmit = e => {
    e && e.preventDefault();
    console.log("content", content);
  };

  const onInputChange = (inputName, inputValue) => {
    setContent(inputValue);
  };
  return (
    <form onSubmit={onSubmit}>
      <InputField
        label={label}
        className='add-todo'
        inputType='text'
        inputName='addTodo'
        onInputChange={onInputChange}
      />
    </form>
  );
};

export default AddTodo;
