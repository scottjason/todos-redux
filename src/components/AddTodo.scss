// .add-todo {
//   position: relative;
//   display: flex;
//   align-items: center;
//   margin: 0 auto;
//   width: 100%;
//   min-width: 400px;
//   max-width: 650px;
//   height: auto;
//   min-height: 50px;
//   padding: 1rem;
//   margin-bottom: 0.5rem;
//   background-color: #46529d;
//   border: 1px solid rgba(225, 225, 225, 0.5);
// }
