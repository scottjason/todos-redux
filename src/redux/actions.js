import { ADD_TODO, EDIT_TODO } from './actionTypes';

export const addTodo = text => ({
  type: ADD_TODO,
  payload: {
    text,
  },
});

export const editTodo = todo => ({
  type: EDIT_TODO,
  payload: {
    todo,
  },
});
