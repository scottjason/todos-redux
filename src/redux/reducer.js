import { ADD_TODO, EDIT_TODO } from './actionTypes';
import { v4 as uuidv4 } from 'uuid';

const initialState = [
  {
    _id: '9b1deb4d-3b7d-4bad-9bdd-2b0d7b3dcb6d',
    text: 'pick up groceries',
    createtAt: Date.now(),
    isReadOnly: true,
    isComplete: false,
  },
  {
    _id: '8b1deb4d-3b7d-4bad-9bdd-2b0d7b3dcb6d',
    text: 'fold the laundry',
    createtAt: Date.now(),
    isReadOnly: true,
    isComplete: false,
  },
];

export const generateTodo = text => {
  return {
    _id: uuidv4(),
    text: text,
    createtAt: Date.now(),
    isReadOnly: true,
    isComplete: false,
  };
};

export const todosReducer = (state = initialState, action) => {
  switch (action.type) {
    case ADD_TODO: {
      return [...state, generateTodo(action.payload.text)];
    }
    case EDIT_TODO: {
      return state.map(todo => {
        if (action.payload.todo._id === todo._id) {
          return { ...todo, text: todo.text };
        }
        return todo;
      });
    }
    default: {
      return state;
    }
  }
};
