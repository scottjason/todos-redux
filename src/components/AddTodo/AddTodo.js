import React, { useState } from 'react';
import { connect } from 'react-redux';
import { addTodo } from '../../redux/actions';
import InputField from '../shared/InputField';

const AddTodo = props => {
  const [text, setText] = useState('');
  
  const label = {
    text: 'add to do',
    htmlFor: 'add-to-do',
  };

  const onInputChange = inputValue => setText(inputValue);

  const onSubmit = e => {
    e && e.preventDefault();
    props.addTodo(text);
    setText('');
  };

  return (
    <form onSubmit={onSubmit}>
      <InputField
        label={label}
        className='add-todo'
        inputType='text'
        inputName='addTodo'
        value={text}
        isReadOnly={false}
        onInputChange={onInputChange}
      />
    </form>
  );
};

const mapStateToProps = state => {
  const { todos } = state;
  return { todos };
};

const mapDispatchToProps = dispatch => {
  return {
    addTodo: text => dispatch(addTodo(text)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(AddTodo);
