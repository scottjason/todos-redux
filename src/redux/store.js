import { createStore, combineReducers } from 'redux';
import { todosReducer } from './reducer';

const reducers = combineReducers(todosReducer)

export default createStore(
  reducers,
  window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__()
);
