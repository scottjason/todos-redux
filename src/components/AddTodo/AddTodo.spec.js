import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { addTodo } from '../../redux/actions';
import { ADD_TODO } from '../../redux/actionTypes';
import AddTodo from './AddTodo';

const middleware = [];
const initialState = {};
const mockStore = configureStore(middleware);
const store = mockStore(initialState);

describe('AddTodo', () => {
  
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <AddTodo />
        </Provider>
      ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('dispatches the action for adding a todo', () => {
    const text = 'walk the dog';
    store.dispatch(addTodo(text));

    const actions = store.getActions();
    const expectedPayload = {
      type: ADD_TODO,
      payload: {
        text: text,
      },
    };
    expect(actions).toEqual([expectedPayload]);
  });
});
