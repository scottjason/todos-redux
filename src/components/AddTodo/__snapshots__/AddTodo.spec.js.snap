// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddTodo renders correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <label
    className="visually-hidden"
    htmlFor="add-to-do"
  >
    add to do
  </label>
  <input
    autoComplete="off"
    className="add-todo"
    id="add-to-do"
    name="addTodo"
    onChange={[Function]}
    readOnly={false}
    type="text"
    value=""
  />
</form>
`;
